### PR TITLE
Parity between UI and CLI

### LINKED ISSUE #920: Feature parity between UI and CLI
**Feature description**

There are currently a few features that are only supported via UI but should be available in the CLI. Among them 

- updating
- getting summary
- setting config values 

These should be implemented in the CLI. Exact details are to be determined.

### PR DESCRIPTION
**Pull request by mrpalide**
Did you run `make format && make check`? Yes

Fixes #920

 Changes:	
- Add `summary` command as `skywire-cli visor summary`
- Add `update` command as `skywire-cli visor update`
- Add `BUILDTAG` to skywire-cli on goreleaser.

**How to test this PR**:
- **summary**: 
  - Run `skywire-cli visor summary`.
- **update**: 
  - Build with skybian *BUILDTAG* flag `make build BUILDTAG=skybian`.
  - Run `skywire-cli visor update` to update visor.
  - (Because of no *BUILDTAG* set on skywire-cli v0.5.1) Build with skybian *BULDTAG* again.
  - Run `skywire-cli visor update` to get message about visor has the latest version.
  - For get error on non-skybian *BUILDTAG*, make build without flag and run update command. Now should get unable to update because of non-skybian *BUILDTAG*.
- **config**
  - Multiple flags added to change their values on config file. Listed as bellow:
    - `vpn-client-killswitch`: could be *true* or *false*.
    - `add-vpn-client-server`: add vpn server address to vpn-client.
    - `add-vpn-client-passcode`: add passcode of vpn server to vpn-client.
    - `reset-vpn-client`: reset all config of vpn-client.
    - `add-vpn-server-passcode`: add passcode for vpn-server.
    - `vpn-server-secure`: could be *true* or *false*.
    - `reset-vpn-server`: reset all config of vpn-server.
    - `add-skysocks-client-server`: add skysocks server pk to skysocks-client.
    - `reset-skysocks-client`: reset skysocks-client config.
    - `add-skysocks-passcode`: set passcode for skysocks server.
    - `reset-skysocks`: reset skysocks server config.
    - `set-public-autoconnect`: could be *true* or *false*.
    - `set-minhop`: should be an integer number greater equal 0.

### CODE REVIEW

**Comment by 0pcom** [comment]
Nice work, happy to see this PR

**Comment by mrpalide** [comment]
> Nice work, happy to see this PR

Please review this, and posted me about your opinion or needed commands.

File: .goreleaser.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ builds:
     env:
       - CGO_ENABLED=0
     main: ./cmd/skywire-cli/
-    ldflags: -s -w -X github.com/skycoin/dmsg/buildinfo.version=v{{.Version}} -X github.com/skycoin/dmsg/buildinfo.commit={{.ShortCommit}} -X github.com/skycoin/dmsg/buildinfo.date={{.Date}} 
+    ldflags: -s -w -X github.com/skycoin/dmsg/buildinfo.version=v{{.Version}} -X github.com/skycoin/dmsg/buildinfo.commit={{.ShortCommit}} -X github.com/skycoin/dmsg/buildinfo.date={{.Date}} -X github.com/skycoin/skywire/pkg/visor.BuildTag=skybian
```
Comment (alexadhy) [inline .goreleaser.yml]: the skybian tag should not be here for all releases, I think macOS and amd64 variant of Linux aren't part of skybian right?

**Review by alexadhy — CHANGES_REQUESTED** [review]
Need some changes at how to add skybian flags

File: .goreleaser.yml (hunk 1 above)
Comment (jdknives) [reply .goreleaser.yml]: @alexadhy you are right I guess. They are not. Can we use the build tag `$OS-binary` for them?

Comment (mrpalide) [reply .goreleaser.yml]: We should separate them on goreleaser, right?

**Comment by 0pcom** [comment]
> > Nice work, happy to see this PR
> 
> Please review this, and posted me about your opinion or needed commands.

I need the dmsgpty-cli terminal as an available subcommand in skywire-cli to have an alternative to a ssh session in order to access the visors and do manual updates, via skywire. This should also be good for general troubleshooting of running visors, right?

I know that there more to do in dmsg repo before this can happen, as @jdknives had pointed out before.

the dmsgpty-ui is already implemented in the hypervisor interface as the 'full terminal', so it seems that the dmsgpty-cli could be in skywire-cli similarly..

**Comment by mrpalide** [comment]
**Updates**:
- add `x` flag to `conifg gen` for use old hypervisors list in new config.
- add `hvpk` to get pubkeys from config by `skywire-cli visor hvpk`
- for both `pk` and `hvpk` on `skywire-cli visor`, if set `-i` flag as path of config, will return directed from file and not use rpc request to online visor. It is usable for offline visor.

File: .goreleaser.yml (hunk 1 above)
Comment (jdknives) [reply .goreleaser.yml]: https://goreleaser.com/customization/build/#build-hooks Lets use the `.Target` variable and we will separate whether or not the target is Skybian in the frontend.

File: cmd/skywire-cli/commands/config/gen.go [hunk 2]
```diff
@@ -22,20 +22,22 @@ func init() {
...
+	skybianConfig      bool
+	hypervisor         bool
+	hypervisorPKs      string
 )
 
 func init() {
 	genConfigCmd.Flags().Var(&sk, "sk", "if unspecified, a random key pair will be generated.\n")
 	genConfigCmd.Flags().StringVarP(&output, "output", "o", "skywire-config.json", "path of output config file.")
 	genConfigCmd.Flags().BoolVarP(&replace, "replace", "r", false, "rewrite existing config (retains keys).")
+	genConfigCmd.Flags().BoolVarP(&replaceHypervisors, "use-old-hypervisors", "H", false, "use old hypervisors keys.")
```
Comment (jdknives) [inline cmd/skywire-cli/commands/config/gen.go]: Why not a lowercase `h` here? We dont use capitalized flags anywhere else here.

File: cmd/skywire-cli/commands/config/update.go [hunk 3]
```diff
@@ -21,11 +21,24 @@ func init() {
...
+	resetVPNclient         bool
+	addVPNServerPasscode   string
+	setVPNServerSecure     string
+	resetVPNServer         bool
+	addSkysocksClientSrv   string
+	resetSkysocksClient    bool
+	skysocksPasscode       string
+	resetSkysocks          bool
+	setPublicAutoconnect   string
+	minhops                int
```
Comment (jdknives) [inline cmd/skywire-cli/commands/config/update.go]: Please capitalize `Hops`.

File: cmd/skywire-cli/commands/config/update.go [hunk 4]
```diff
@@ -34,6 +47,25 @@ func init() {
...
+	updateConfigCmd.Flags().StringVar(&addVPNClientSrv, "add-vpn-client-server", "", "add server address to vpn-client")
+	updateConfigCmd.Flags().StringVar(&addVPNClientPasscode, "add-vpn-client-passcode", "", "add passcode of server if needed")
+	updateConfigCmd.Flags().BoolVar(&resetVPNclient, "reset-vpn-client", false, "reset vpn-client's configurations")
+
+	updateConfigCmd.Flags().StringVar(&addVPNServerPasscode, "add-vpn-server-passcode", "", "add passcode to vpn-server")
+	updateConfigCmd.Flags().StringVar(&setVPNServerSecure, "vpn-server-secure", "", "change secure mode status of vpn-server")
+	updateConfigCmd.Flags().BoolVar(&resetVPNServer, "reset-vpn-server", false, "reset vpn-server's configurations")
+
+	updateConfigCmd.Flags().StringVar(&addSkysocksClientSrv, "add-skysocks-client-server", "", "add skysocks server address to skysock-client")
+	updateConfigCmd.Flags().BoolVar(&resetSkysocksClient, "reset-skysocks-client", false, "reset skysocks-client's configuration")
```
Comment (jdknives) [inline cmd/skywire-cli/commands/config/update.go]: `'s` can be removed.

File: cmd/skywire-cli/commands/config/update.go [hunk 5]
```diff
@@ -34,6 +47,25 @@ func init() {
 	updateConfigCmd.Flags().StringVarP(&environment, "environment", "e", "production", "desired environment (values production or testing)")
 	updateConfigCmd.Flags().StringVar(&addHypervisorPKs, "add-hypervisor-pks", "", "public keys of hypervisors that should be added to this visor")
 	updateConfigCmd.Flags().BoolVar(&resetHypervisor, "reset-hypervisor-pks", false, "resets hypervisor`s configuration")
+
+	updateConfigCmd.Flags().StringVar(&setVPNClientKillswitch, "vpn-client-killswitch", "", "change killswitch status of vpn-client")
```
Comment (jdknives) [inline cmd/skywire-cli/commands/config/update.go:51]: Why is that a string istead of a bool? Same question for the other values we set as bools internally such as `autoconect` and `secure` value

Comment (jdknives) [reply cmd/skywire-cli/commands/config/update.go:51]: Is it because you want to avoid the boolean default value?

**Review by jdknives — CHANGES_REQUESTED** [review]

File: cmd/skywire-cli/commands/config/update.go (hunk 5 above)
Comment (mrpalide) [reply cmd/skywire-cli/commands/config/update.go:51]: Yes, was related to default value problem. Cobra has `isChanged()` method, but tbh I couldn't use it correctly, so I use string instead.

File: cmd/skywire-cli/commands/config/update.go (hunk 3 above)
Comment (mrpalide) [reply cmd/skywire-cli/commands/config/update.go]: Done.

File: cmd/skywire-cli/commands/config/gen.go (hunk 2 above)
Comment (mrpalide) [reply cmd/skywire-cli/commands/config/gen.go]: `h` is default flag for help command. I change `H` to `x` here.

**Review by jdknives — APPROVED** [review]

**Comment by ersonp** [comment]
Getting this panic
```
erson@erson-69:~/Workspace/Go/skywire$ ./skywire-cli config update --add-vpn-client-server 0357414c43a92075758ed889ec29997e2c16ba34bf873f76e8dae2edd7b9b97122
[2021-11-08T15:23:29+05:30] INFO [visor:config]: Flushing config to file. config_version="v1.1.0" filepath="skywire-config.json"
panic: runtime error: index out of range [3] with length 0

goroutine 1 [running]:
github.com/skycoin/skywire/cmd/skywire-cli/commands/config.changeAppsConfig(0xc000312000, 0x9cf67f, 0xa, 0x9ccc14, 0x5, 0xc0002a8a00, 0x42)
        /home/erson/Workspace/Go/skywire/cmd/skywire-cli/commands/config/update.go:216 +0x3a5
github.com/skycoin/skywire/cmd/skywire-cli/commands/config.glob..func4(0xe0fbc0, 0xc000284500, 0x0, 0x2)
        /home/erson/Workspace/Go/skywire/cmd/skywire-cli/commands/config/update.go:136 +0xe65
github.com/spf13/cobra.(*Command).execute(0xe0fbc0, 0xc0002844e0, 0x2, 0x2, 0xe0fbc0, 0xc0002844e0)
        /home/erson/Workspace/Go/skywire/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xe0f3e0, 0xc000000180, 0xc00013df78, 0x407f25)
        /home/erson/Workspace/Go/skywire/vendor/github.com/spf13/cobra/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/erson/Workspace/Go/skywire/vendor/github.com/spf13/cobra/command.go:887
github.com/skycoin/skywire/cmd/skywire-cli/commands.Execute()
        /home/erson/Workspace/Go/skywire/cmd/skywire-cli/commands/root.go:30 +0x31
main.main()
        /home/erson/Workspace/Go/skywire/cmd/skywire-cli/skywire-cli.go:8 +0x25
```
Whenever commands are used in the following order
```
./skywire-cli config update --vpn-client-killswitch true
```
```
./skywire-cli config update --add-vpn-client-server 031a5d05797d18ecf13e6f29029b0c27a1a26117467a98ff21df70cc3ddf7a20f2
```
```
./skywire-cli config update --vpn-client-killswitch false
```
```
./skywire-cli config update --add-vpn-client-server 0357414c43a92075758ed889ec29997e2c16ba34bf873f76e8dae2edd7b9b97122
```

**Comment by ersonp** [comment]
No error is given when
```
./skywire-cli config update --vpn-client-killswitch asdsad
```
anything other than `true` or `false` is given.

**Comment by ersonp** [comment]
```
./skywire-cli config update --vpn-server-secure true
```
```
./skywire-cli config update --add-vpn-server-passcode testpass
```
not reflected in config
```json
{
     "name": "vpn-server",
     "args": [
     "--passcode",
     ""
     ],
     "auto_start": false,
     "port": 44
}
```

**Review by ersonp — APPROVED** [review]

**Pull request merged**